src/client.rs

```diff
@@ -426,34 +426,35 @@ impl Client {
     /// # Example
     ///
     /// ```no_run
-    /// use kube::api::ListParams;
     /// use std::time::Duration;
     /// use tokio::time::error::Elapsed;
+    /// use kube::runtime::watcher;
     /// use k8s_openapi::api::core::v1::Pod;
     /// use stackable_operator::client::{Client, create_client};
     ///
     /// #[tokio::main]
     /// async fn main(){
+    ///
     /// let client: Client = create_client(None).await.expect("Unable to construct client.");
-    /// let lp: ListParams =
-    ///         ListParams::default().fields(&format!("metadata.name=nonexistent-pod"));
+    /// let watcher_config: watcher::Config =
+    ///         watcher::Config::default().fields(&format!("metadata.name=nonexistent-pod"));
     ///
     /// // Will time out in 1 second unless the nonexistent-pod actually exists
     ///  let wait_created_result: Result<(), Elapsed> = tokio::time::timeout(
     ///          Duration::from_secs(1),
-    ///          client.wait_created::<Pod>(&client.default_namespace, lp.clone()),
+    ///          client.wait_created::<Pod>(&client.default_namespace, watcher_config),
     ///      )
     ///      .await;
     /// }
     /// ```
     ///
-    pub async fn wait_created<T>(&self, namespace: &T::Namespace, lp: ListParams)
+    pub async fn wait_created<T>(&self, namespace: &T::Namespace, watcher_config: watcher::Config)
```

Comment:
How about not changing the signature so as to not introduce an unnecessary incompatibility?

Reply:
I dont understand?

The watcher now gets a different struct (granted, some fields are similar) but there are differences like https://github.com/kube-rs/kube/blob/e01187e13ba364ccecec452e023316a62fb13e04/kube-runtime/src/watcher.rs#L206 that should be configurable?

You just want to use the Label / Field selectors from the ListParams? We will be missing config options by doing this.